src/ACEtk/Xsdir/src/stream.hpp

```
@@ -0,0 +1,36 @@
+/**
+ *  @brief operator>> for an xsdir
+ *
+ *  @param[in] in       the input stream
+ *  @param[in] entry    the xsdir
+ *
+ *  @return the input stream
+ */
+inline std::istream &operator>>( std::istream& in, Xsdir& xsdir ) {
+
+  try {
+
+    Xsdir temp( in );
+    xsdir = std::move( temp );
+  }
+  catch ( ... ) {
+
+    // nothing to do to recover from the error: in is already in fail state
```

Does an empty catch block act differently than just not having a try/catch at all?

---

Streaming is particular in that it does not throw exceptions, it actually changes the stream state which a user can then check (this is how stream operators in the standard library work). I catch any exception here to mimic that behaviour. The stream state is already set to fail in the constructor that threw the exception, but I might just do it here as well to make it clear that we do this to mimic the standard library behaviour.